Core/WebViewController.swift

```diff
@@ -239,12 +243,17 @@ open class WebViewController: UIViewController {
         webView.goForward()
     }
 
-    public func tearDown() {
+    fileprivate func removeObservers() {
         webView.removeObserver(self, forKeyPath: #keyPath(WKWebView.estimatedProgress))
         webView.removeObserver(self, forKeyPath: #keyPath(WKWebView.hasOnlySecureContent))
         webView.removeObserver(self, forKeyPath: #keyPath(WKWebView.url))
         webView.removeObserver(self, forKeyPath: #keyPath(WKWebView.canGoForward))
         webView.removeObserver(self, forKeyPath: #keyPath(WKWebView.canGoBack))
+    }
+    
+    public func tearDown() {
+        addObservers() // prevents a crash if the observers were not there for some reason (e.g. WKWebView process crashed)
```

Comment:
Did you actually observe that? I've been running this on my device since I made the change and haven't seen that. I stepped over and memory profiled that code as well.

Reply:
Oh interesting, I'm only seeing the crash on iOS 10, Apple must have fixed it for iOS 11. Good move Apple, I always found it annoying, especially since there's no way to "check" if a KVO observer is set.

Reply:
I was just downloading a 10.x simulator to test it on... 🙄

Reply:
OK, just pushed a commit that wraps this `addObservers` in a `if #available`

Reply:
Let's grab a zoom call and dig into this more on Monday.

Reply:
tbh, this was supposed to be a quick harmless speculative fix - if it's anything more than that (and it already is) then I've failed to achieve the objective.  Better to can it and try to replicate, but that'll take time so probably needs spinning up as a project.